packages/wrangler/src/deploy/config-diffs.ts

```diff
@@ -366,39 +384,99 @@ function populateConfigPatchArray(diff: JsonLike[], patchArray: JsonLike[]) {
 	elementsToAppend.forEach((el) => patchArray.push(el));
 }
 
+/**
+ * Given a key discerns wether the key is only allowed at the top level of a config object
+ *
+ * @param key The target key
+ * @returns true is the config is top-level only, false otherwise
+ */
```

Comment:
Do we actually need this logic? The top level keys seem to be:
```
"legacy_env" | "send_metrics" | "dev" | "site" | "wasm_modules" | "text_blobs" | "data_blobs" | "alias" | "keep_vars"
```
and the constructed remote config (from `convertWorkerToWranglerConfig()`) doesn't insert any of those, I don't think.